Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -1337,4 +1341,37 @@ HTTPServer server({
 }
 #endif
 
+TEST(SiteIsolation, SetFocusedFrame)
+{
+    auto mainframeHTML = "<script>"
+    "    function setFocus() {"
+    "        document.querySelector('iframe').focus();"
+    "        alert('focused')"
+    "    }"
+    "</script>"
+    "<iframe onload='setFocus()' src='https://domain2.com/subframe'></iframe>"_s;
+
+    auto subframeHTML = ""_s;
+
+    HTTPServer server({
+        { "/mainframe"_s, { mainframeHTML } },
+        { "/subframe"_s, { subframeHTML } }
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
+    [navigationDelegate allowAnyTLSCertificate];
+    auto configuration = server.httpsProxyConfiguration();
+    enableSiteIsolation(configuration);
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration]);
+    webView.get().navigationDelegate = navigationDelegate.get();
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://domain1.com/mainframe"]]];
+    EXPECT_WK_STREQ([webView _test_waitForAlert], "focused");
+
+    auto mainframe = WKPageGetMainFrame([webView _pageForTesting]);
```

Comment:
This is currently only available on macOS.  Could we instead add SPI to WKFrameInfo?  BOOL _focused